draft-ietf-httpbis-resumable-upload.md

```diff
@@ -512,6 +512,34 @@ If the client knows the integrity digest of the content from an upload creation
+
+The following subsections describe three different, typical upload strategies that are suited for common environments. Note that these modes are never explicitly communicated to the server and clients are not required to stick to one strategy, but can mix and adapt them to their needs.
+
+## Optimistic Upload Creation
+
+An "optimistic upload creation" can be used if the client knows or assumes with a high degree of confidence that the server supports resumable uploads under the target URI and that the client is capable of handling and processing interim responses. The upload creation request then includes the full file because the client assumes that the file will be transferred without interruptions.
+
+The benefit of this method is that if the upload creation request succeeds, the file was transferred in a single request avoiding additional round trips. A drawback is that the client may be restricted in its ability to resume that upload. If the upload was interrupted before the client received a `104 (Upload Resumption Supported)` intermediate response with the upload URL, the client cannot resume that upload due to the missing upload URL. The intermediate response might not be received if the interruption happens too early in the message exchange, the server does not support sending the `104 (Upload Resumption Supported)` intermediate response, or an intermediary dropped the intermediate response. The client must either ultimately fail the entire upload or retry the upload creation request if this is allowed by the application at the expense of transferring file data again.
+
+The "optimistic upload creation" attempts to upload the file in one request at the potential expense of either not finishing the upload at all or re-transmitting file data again.
+
+## Careful Upload Creation
+
+With a "careful upload creation" the client sends an empty upload creation request without including any file data. Upon successful response reception, the client can use the included upload URL to transmit the file data ({{upload-appending}}) and resume the upload at any stage if an interruption occurs. The retransmission of file data or the ultimate upload failure that can happen with an "optimistic upload creation" is therefore avoided at the expense of an additional request that does not carry file data.
+
+This approach best suited if the client cannot receive intermediate responses, e.g. due to a limitation in the provided HTTP interface, or if large files are transferred where the cost of the additional request is miniscule compared to the effort of transferring the large file itself.
+
+## Upgrading To Resumable Uploads
```

Comment:
This seems indistinguishable from the optimistic approach to me.  The use of Upload-Complete might distinguish it, but I don't see how these are functionally different.

Reply:
You're correct. I merged them.